test/y2storage/blk_device_test.rb

```diff
@@ -527,4 +527,34 @@
       end
     end
   end
+
+  describe ".sorted_by_name" do
+    before { fake_scenario("sorting/disks_and_dasds1") }
+
+    it "returns all the blk devices sorted by name" do
+      devices = Y2Storage::BlkDevice.sorted_by_name(fake_devicegraph)
+      expect(devices.map(&:basename)).to eq %w(
+        dasda dasda1 dasda2 dasda10 dasdb dasdab dasdb1 dasdb2 dasdb3
```

Comment:
I would expect dasdb1 before dasdab.

Reply:
@aschnell That's doable. See https://github.com/yast/yast-storage-ng/pull/434 which works as you expect.

Here is the diff between both https://github.com/yast/yast-storage-ng/pull/434/commits/e3e87907c063df95f3107fe62abfc4f1eb95311f

For the first POC I just wanted to keep it simple to make the code easier to follow (for the sake of the discussion) and, eventually, to translate to C++. But I'm fine with any of the alternatives.

I can try to merge any of them into yast-storage-ng (whichever you guys prefer), with the translation to C++ as a separate PBI or extra task for this PBI.